src/internal/pfssync/cache_client.go

```diff
@@ -40,14 +40,14 @@ func (cc *CacheClient) GetFileTAR(commit *pfs.Commit, path string) (io.ReadClose
 	if c, ok := cc.get(key); ok {
 		return cc.APIClient.GetFileTAR(c, path)
 	}
-	id, err := cc.APIClient.GetProjectFileSet(commit.Branch.Repo.Project.GetName(), commit.Branch.Repo.Name, commit.Branch.Name, commit.ID)
+	id, err := cc.APIClient.GetProjectFileSet(commit.Repo.Project.GetName(), commit.Repo.Name, commit.Branch.Name, commit.ID)
 	if err != nil {
 		return nil, err
 	}
 	if err := cc.renewer.Add(cc.APIClient.Ctx(), id); err != nil {
 		return nil, err
 	}
-	commit = client.NewProjectCommit(commit.Branch.Repo.Project.GetName(), client.FileSetsRepoName, "", id)
+	commit = client.NewProjectCommit(commit.Repo.Project.GetName(), client.FileSetsRepoName, "", id)
```

Comment:
I feel that new code we write should probably not use the convenience constructors in package `client`, and should instead either construct messages explicitly or use constructors in a new internal package.